src/wallet/coinselection.cpp

```diff
@@ -395,3 +395,53 @@ CAmount GetSelectionWaste(const std::set<CInputCoin>& inputs, CAmount change_cos
 
     return waste;
 }
+
+void SelectionResult::ComputeAndSetWaste(CAmount change_cost)
+{
+    m_waste = GetSelectionWaste(m_selected_inputs, change_cost, m_target, m_use_effective);
+}
+
+CAmount SelectionResult::GetWaste() const
+{
+    Assume(m_waste != std::nullopt);
+    return *m_waste;
```

Comment:
In 1b1ad425d6b2ca951e0a7e6096819fc5e468af4b:

Would it be better if, when waste is `nullopt`, calculate it? The `*m_waste` would throw an exception if `m_waste == std::nullopt`, so the `Assume` doesn't do much.

Reply:
The cost of change is not known to `SelectionResult` so it cannot be calculated.

Reply:
ah right 🤦